sources/tools/Stride.Importer.Assimp/MeshConverter.cs

```diff
@@ -0,0 +1,1364 @@
+                var specularModel = new MaterialSpecularMicrofacetModelFeature
+                {
+                    Fresnel = new MaterialSpecularMicrofacetFresnelSchlick(),
+                    Visibility = new MaterialSpecularMicrofacetVisibilityImplicit(),
+                    NormalDistribution = new MaterialSpecularMicrofacetNormalDistributionBlinnPhong()
+                };
+                finalMaterial.Attributes.SpecularModel = specularModel;
+            }
+            else if (textureType == TextureType.TextureTypeEmissive)
+            {
+                // TODO: Add support
+            }
+            else if (textureType == TextureType.TextureTypeAmbient)
+            {
+                // TODO: Add support
+            }
+            else if (textureType == TextureType.TextureTypeReflection)
+            {
+                // TODO: Add support
+            }
+            if (textureType == TextureType.TextureTypeOpacity)
+            {
+                // TODO: Add support
+            }
+            else if (textureType == TextureType.TextureTypeShininess)
+            {
+                // TODO: Add support
+            }
+            if (textureType == TextureType.TextureTypeSpecular)
+            {
+                // TODO: Add support
+            }
+            else if (textureType == TextureType.TextureTypeNormals)
+            {
+                finalMaterial.Attributes.Surface = new MaterialNormalMapFeature(computeColorNode);
+            }
+            else if (textureType == TextureType.TextureTypeDisplacement)
+            {
+                // TODO: Add support
+            }
+            else if (textureType == TextureType.TextureTypeHeight)
+            {
+                // TODO: Add support
+            }
+        }
+
+        private unsafe IComputeColor GenerateOneTextureTypeLayers(Silk.NET.Assimp.Material* pMat, TextureType textureType, int textureCount, MaterialAsset finalMaterial)
+        {
+            var stack = Material.Materials.ConvertAssimpStackCppToCs(assimp, pMat, textureType);
+
+            var compositionFathers = new Stack<IComputeColor>();
+
+            var sets = new Stack<int>();
+            sets.Push(0);
+
+            var nbTextures = assimp.GetMaterialTextureCount(pMat, textureType);
+
+            IComputeColor curComposition = null, newCompositionFather = null;
+
+            var isRootElement = true;
+            IComputeColor rootMaterial = null;
+
+            while (!stack.IsEmpty)
+            {
+                var top = stack.Pop();
+
+                IComputeColor curCompositionFather = null;
+                if (!isRootElement)
+                {
+                    if (compositionFathers.Count == 0)
+                        Logger.Error("Texture Stack Invalid : Operand without Operation.");
+
+                    curCompositionFather = compositionFathers.Pop();
+                }
+
+                var set = sets.Pop();
```

Comment:
This one is removed too early, if the current material element is not result of an operation there's no need to pop this one here